<!DOCTYPE html>
<html>

<head>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-98278907-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-98278907-1');
</script>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Customizing the default BayesPI-BAR2 pipeline</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
</head>

<body class="stackedit">
  <div class="stackedit__html"><h1 id="customizing-the-default-bayespi-bar2-pipeline">Customizing the default BayesPI-BAR2 pipeline</h1>
<p>The <code>bayespi_bar2_pipeline.py</code> script is the starting point for users wishing to use <strong>BayesPI-BAR2</strong> to process their own datasets. It shows a reasonable sequence of execution of the main package tools to obtain a final result. The source code in the file is simple, well-commented and should be easy to modify. The pipeline, as it is implemented in this script, can be logically divided into 8 steps, each step being a call to a tool from the package:</p>
<ol>
<li>Compute differentially expressed genes between patient group and the normal cell group (here we use gene expression in melanocytes as the normal group). Calls <code>differential_expression.py</code>.</li>
<li>Select <em>regions of interest</em>, here defined as regions near transcription start sites of differentially expressed genes. Calls <code>gene_regions.py</code>.</li>
<li>Compute mutation blocks in the regions of interest using <strong>MuSSD</strong>. Calls <code>mussd.py</code>.</li>
<li>Select those blocks that are mutated more than average. The melanoma dataset contains large number of mutations, leading to many identified mutation blocks. This step removes 90% of blocks that have relatively few mutations. In cases when there are few mutation blocks identified by <strong>MuSSD</strong>, this step is probably unnecessary. Calls <code>highly_mutated_blocks.py</code>.</li>
<li>Compute affinity changes for the foreground (patient) mutation blocks using <strong>BayesPI-BAR</strong>. Calls <code>bayespi_bar.py</code>.</li>
<li>Compute the random background model for affinity changes, matching its parameters to the chosen foreground blocks. Calls <code>background_affinity_changes.py</code>. There is a choice of background models:
<ul>
<li>A tumor-derived background model will randomly samples patients’ tumor somatic mutations to build the background model. It assumes that most mutations have no effect. This model most closely matches various features of the tumor mutation biases, including location-specific ones. However it requires a large mutation sample to generate a good background model. At least a few thousand mutations is needed.</li>
<li>A mutation signature model uses k-mer probability distribution specified by the user to randomly generate mutations. These distributions for many known cancer subtypes can be obtained from COSMIC, or generated by other means. The mutation locations are chosen randomly.</li>
<li>A simplest uninformed background model will generate mutations uniformly, in both location and nucleotide replacement. False positives due to cancer-specific mutation biases are to be expected with this model.</li>
</ul>
</li>
<li>Run the affinity change significance tests for all foreground blocks vs. the background model. Calls <code>affinity_change_significance_test.py</code>.</li>
<li>Remove the transcription factors that are not expressed in patient samples from the final predictions. This is implemented as an auxiliary tool from the demo folder because it relies on the specific naming convention of the supplied PWM database.</li>
</ol>
</div>
</body>

</html>
